tests/phpunit/Mutation/PrioritizedVisitorsNodeTraverserTest.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutation;
+
+use DomainException;
+use Generator;
+use Infection\Mutation\PrioritizedVisitorsNodeTraverser;
+use Infection\Tests\Fixtures\Mutation\FakeNodeTraverser;
+use Infection\Tests\Fixtures\PhpParser\FakeVisitor;
+use Infection\Tests\Fixtures\PhpParser\RecordedVisitor;
+use InvalidArgumentException;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor;
+use PHPUnit\Framework\TestCase;
+
+final class PrioritizedVisitorsNodeTraverserTest extends TestCase
+{
+    /**
+     * @var PrioritizedVisitorsNodeTraverser
+     */
+    private $traverser;
+
+    protected function setUp(): void
+    {
+        $this->traverser = new PrioritizedVisitorsNodeTraverser(new FakeNodeTraverser());
+    }
+
+    public function test_it_cannot_add_visitors_without_a_priority(): void
+    {
+        try {
+            $this->traverser->addVisitor(new FakeVisitor());
+
+            $this->fail('Expected an exception to be thrown.');
+        } catch (DomainException $exception) {
```

Comment:
How about using expectException, and expectExceptionMessage here?

Reply:
I really don't like the expect exception because it hides what exact statement is expected to fail.

In other cases where you want to check the status code or previous, the PHPUnit API for it makes it more awkward than anything IMO. I also recall Sebastian wanting to removing at some point but it would just be too much of a BC break for PHPUnit